corehq/apps/hqwebapp/management/commands/build_requirejs.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import print_function
+import json
+import os
+import re
+import yaml
+from django.contrib.staticfiles import finders
+from django.conf import settings
+from subprocess import call
+
+from corehq.apps.hqwebapp.management.commands.resource_static import Command as ResourceStaticCommand
+
+
+class Command(ResourceStaticCommand):
+    help = '''
+        Runs RequireJS optimizer to concatenate, minify, and bundle JavaScript files
+        and set them up with the CDN.
+    '''
+
+    root_dir = settings.FILEPATH
+
+    def handle(self, **options):
+        try:
+            from resource_versions import resource_versions
+        except (ImportError, SyntaxError):
+            resource_versions = {}
+
+        # Write build.js file to feed to r.js
+        with open(os.path.join(self.root_dir, 'staticfiles', 'hqwebapp', 'yaml', 'requirejs.yaml'), 'r') as f:
+            config = yaml.load(f)
+
+            bundles = {}
+            all_modules = []
+            prefix = os.path.join(os.getcwd(), 'corehq')
+            for finder in finders.get_finders():
+                if isinstance(finder, finders.AppDirectoriesFinder):
+                    for path, storage in finder.list(['.*', '*~', '* *', '*.*ss', '*.png']):
+                        if not storage.location.startswith(prefix):
+                            continue
+                        if path.endswith(".js"):
+                            directory = re.sub(r'/[^/]*$', '', path)
+                            if directory not in bundles:
+                                bundles[directory] = []
+                            bundles[directory].append(path[:-3])
+                            all_modules.append(path[:-3])
+
+            customized = {re.sub(r'/[^/]*$', '', m['name']): True for m in config['modules']}
+            for directory, inclusions in bundles.iteritems():
+                if directory not in customized and not directory.startswith("app_manager/js/vellum"):
+                    # Add this module's config to build config
+                    config['modules'].append({
+                        'name': os.path.join(directory, 'bundle'),
+                        'include': inclusions,
+                        'excludeShallow': [name for name in all_modules if name not in inclusions],
+                        'exclude': ['hqwebapp/js/common'],
+                    })
+
+            # Write .js files to staticfiles
+            for module in config['modules']:
+                with open(os.path.join(self.root_dir, 'staticfiles', module['name'] + ".js"), 'w') as fout:
+                    fout.write("define([], function() {});")
+
+            with open(os.path.join(self.root_dir, 'build.js'), 'w') as fout:
+                fout.write("({});".format(json.dumps(config, indent=4)))
+
+        call(["node", "bower_components/r.js/dist/r.js", "-o", "build.js"])
```

Comment:
where is this `build.js` file created? guessing it's the root of the repository which is already littered with junk. Could we put it in `staticfiles` or somewhere else?

Reply:
moved to `staticfiles`